content/include/function_example_tests.md

```diff
@@ -0,0 +1,59 @@
+    // server_internal_test.go
+    func TestServer_GetHeightForBlock(t *testing.T) {
+        blockID := mocks.GenericHeader.ID()
+        tests := []struct {
+            name string
+            req *GetHeightForBlockRequest
+            mockErr error
+            wantBlockID flow.Identifier
+            checkErr require.ErrorAssertionFunc
+        }{
+            {
+                name: "nominal case",
+                req: &GetHeightForBlockRequest{
+                    BlockID: mocks.ByteSlice(blockID),
+                },
+                mockErr: nil, 
+                wantBlockID: blockID,
+                checkErr: require.NoError,
+            },
+            {
+                name: "handles invalid block request",
+                req: &GetHeightForBlockRequest{},
+                checkErr: require.Error,
+            },
+            {
+                name: "handles failure to retrieve block height from index",
+                req: &GetHeightForBlockRequest{
+                    BlockID: mocks.ByteSlice(blockID),
+                },
+                mockErr: mocks.GenericError,
+                checkErr: require.Error,
+            },
+        }
+
+        for _, test := range tests {
+            test := test
+            t.Run(test.name, func (t *testing.T) {
+                t.Parallel()
```

Comment:
The only case where we should not use parallelization is when testing something that is not concurrency safe. There is zero downside to doing it otherwise, so we should always do it IMO 🤷